### ISSUE TITLE
No output showing for code

### ISSUE THREAD

**Issue by Mohitkumar6122** [psf/black#1720]
**Black v20.8b1**

[Playground link](https://black.now.sh/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4A6EBLldAD2IimZxl1N_Wg0_QLDSluJRcxyy_agGIjCIe-3gOSwQBIue2SQ3AVGKgb6DpEvXgFxOomSVNZSryhnDrctB0C21EbQYxJiJl3l6TKrmVg8q99KqCURDEvtFAY_YTLaJbedZBLdo-YY9UuFkORjX1gbEAPWRwfjy63OrTf4kSlgGppJ1bfHpnVaXywOgsgUN_SuHEH7LvXzHv4al8FFSWuKyEHGhWV63vJkY8W_9mw5dhCaSpBD2Gc3btI--gENFlpwqErnVJubrUuUanjtlbi2nKNG_Y6YRSDnb-jn30QL8yxmFf9gdKPHgsxcBQZgqioT-ctlr-L5J7LY9Ev4sicH_8-hx0Eb0hFbKTV2St9e4SNjccuICW1PrqbdXCdQEPYVg3rMTHQs4kVOsYGSxSBleerilxiPtswQux8UgbXTOLpTZgZ41r3bseMSuK3O51s0Ju33vaisPlydbL5I4UxsQmmDaq4_E4ygVBHzFKiFNvj9Hp-ac-UbzHrX8Q1677uKSSs8qw_GHDB7iphd8wd0TFCRhToV212oe7Wt2oamuYtL9e2aKYwFWOniiUIkfb-BH24JClKFDz4zzO7Hm5QotfONsAaMFPADUSYH8aufObu-QC_AVKA68saZaBKDO8qo7eBSn6C7CiBhE46p7k7qw9z_qbKfVkI5f9hBjNtkMKn6JOXqrchAkJRYlQUM1IF-feirUIeF8B3n80-x6t7p2Mjc2WLXr0aOpYhLBFBTWoRzMuf-Cm_W0ZXnBCE7HCoObHp2Vikrtml_phcSEoPASQ0JCq5_AdycaFpQxMpHEoVa8IihaEKTNIszSGdRjyAkkxFPx5gAbdvduXDcdQRuKI5ue-tVNwZeNI1ZLFjui_7BUx9hpfCxJsP4DavhEFLmtyHAt2QvnUGuEQTUf05mWGaUaKlGnYKpL1YOf7DrQOBmWfEEQ_9hjC8zUmdwOAqbleYTywTl7i0IF0vuzpl2kmCMNeuftqc71BBFnSOOxbmvM3u__OSyQqnRKhrHwSIglqaWAW2LhlYX5WzQguUSyM0q2WczY9ItkzplSQCzRZQFECAhfX8n8_Oc1JSAfnLqDt28fr0X1U1DiU5CKmuUXCSOPqqqEzkbWjClkwoglrHAO3kaRH3j9IHJbXkWq0Fuc5aK4vrPZg2b7EF5-jN3J8rJS7XAdj-3Gb84ci5MxjIjyCrxsF2jAh8AVpwIc5DxfuARYtgCjbSblLJ_CVYwjTiaC038ewymr35qLLgGWYofeo89_1Ks3FU_-MfbWW5fQnrofcfyTcX-PD6cNjwCy1WMm2QLVvUabsZPEeV8lxoad7OJu2fV-SB6TLmST6a4ud6-agYFcVx4RF-EMRihhzktkGnJYGxXrF5LGQCMLpQqpc7qPi1J-wK14GULrJkw_O7MXKlK4WfwoHRT8_Dv-9mr1URGQqJKRnimvR-TLYLO03HVrFGuQHZ0nRN4AUJtDrDkPgOBIwdeGBKgaLmBxwoNfTJFmkD0h9B6P4k6BOtYQ4bAjManFFvgnxivj3JQ64Ag3WjCAzx_CpaqSZ-OKWsvCd4RNLb_LXsvMooSk5QSCb3BspTp2g9jZtV5GZWs0ttXmPPYVvgAAAAC4ifKvPOFflgAB1QmFHQAAGk3PQrHEZ_sCAAAAAARZWg==)

## Options

`--line-length=88`
`--safe`

## Input

```python
#!/usr/bin/env python3

"""
Author : Mohit Kumar
Job Sequencing Problem implemented in python
"""
from collections import namedtuple
from typing import List


class Scheduling:
    def __init__(self, jobs: int) -> None:
        """
        Assign jobs as instance of class Scheduling
        """
        self.jobs = jobs

    def schedule(self, total_jobs: int, deadline: List[int]) -> List[int]:
        """
        Parameteres  : total_jobs  and list of deadline of jobs

        Returns : List of jobs_id which are profitable  and can be done before
                  deadline

        Scheduling(3).schedule(3, [3, 4, 5])
        >>> [1, 2]

        Scheduling(4).schedule(4, [13, 2 33, 16])
        >>> [0, 3, 2]

        """
        self.j = [self.jobs[1]]
        self.x = 2
        while self.x < total_jobs:
            self.k = self.j.copy()
            self.k.append(self.jobs[self.x])
            self.x += 1
            if self.feasible(self.k, deadline):
                self.j = self.k.copy()

        return self.j

    def feasible(self, profit_jobs: List[int], deadline: List[int]) -> bool:
        """
        Parameters : list of current profitable jobs within deadline
                     list of deadline of jobs

        Returns : true if k[-1] job is profitable to us else false

        Scheduling(7).feasible( [6, 5], [0, 1, 2, 2, 3, 3, 4] )
        >>> True

        Scheduling(7).feasible( [6, 5, 1],  [0, 1, 2, 2, 3, 3, 4] )
        >>> False

        """
        self.tmp = profit_jobs
        self.is_feasible = True

        i = 0
        j = 1
        k = 0

        while i < len(self.tmp):
            while j < len(self.tmp):
                self.index1 = self.jobs.index(self.tmp[i])
                self.index2 = self.jobs.index(self.tmp[j])
                j += 1
                if deadline[self.index1] > deadline[self.index2]:
                    (self.tmp[i], self.tmp[j]) = (
                        self.tmp[j],
                        self.tmp[i],
                    )
            i += 1

        while k < len(self.tmp):
            self.job = self.tmp[k]
            self.jobindex = self.jobs.index(self.job)
            self.dlineval = deadline[self.jobindex]
            self.ftest = k + 1
            k += 1
            if self.dlineval < self.ftest:
                self.is_feasible = False
                break
        return self.is_feasible


def main():
    job = namedtuple("job", "job_id deadline profit")
    jobs = [
        job(0, 0, 0),
        job(1, 2, 46),
        job(2, 4, 52),
        job(3, 3, 30),
        job(4, 3, 36),
        job(5, 2, 56),
        job(6, 1, 40),
    ]
    # midresult stores jobs in sorting order of deadline
    midresult = []
    for i in range(len(jobs)):
        current_job = []
        current_job.extend((jobs[i].deadline, jobs[i].profit, jobs[i].job_id))
        midresult.append(current_job)
    midresult.sort(key=lambda k: (k[0], -k[1]))
    (deadline, profit, jobs) = map(list, zip(*midresult))

    scheduling_jobs = Scheduling(jobs)
    scheduled_jobs = scheduling_jobs.schedule(len(jobs), deadline)
    print(f"\n Jobs {scheduled_jobs}")

    finalprofit = []
    finaldl = []

    for i, item in enumerate(scheduled_jobs):
        jobsindex = jobs.index(item)
        finalprofit.append(profit[jobsindex])
        finaldl.append(deadline[jobsindex])

    print(f"\n Profit {finalprofit}")
    print(f"\n Deadline {finaldl}")


if __name__ == "__main__":
    main()

```

## Output

```python

```

## Expected
```
#!/usr/bin/env python3

"""
Author : Mohit Kumar
Job Sequencing Problem implemented in python
"""
from collections import namedtuple
from typing import List


class Scheduling:
    def __init__(self, jobs: int) -> None:
        """
        Assign jobs as instance of class Scheduling
        """
        self.jobs = jobs

    def schedule(self, total_jobs: int, deadline: List[int]) -> List[int]:
        """
        Parameteres  : total_jobs  and list of deadline of jobs

        Returns : List of jobs_id which are profitable  and can be done before
                  deadline

        Scheduling(3).schedule(3, [3, 4, 5])
        >>> [1, 2]

        Scheduling(4).schedule(4, [13, 2 33, 16])
        >>> [0, 3, 2]

        """
        self.j = [self.jobs[1]]
        self.x = 2
        while self.x < total_jobs:
            self.k = self.j.copy()
            self.k.append(self.jobs[self.x])
            self.x += 1
            if self.feasible(self.k, deadline):
                self.j = self.k.copy()

        return self.j

    def feasible(self, profit_jobs: List[int], deadline: List[int]) -> bool:
        """
        Parameters : list of current profitable jobs within deadline
                     list of deadline of jobs

        Returns : true if k[-1] job is profitable to us else false

        Scheduling(7).feasible( [6, 5], [0, 1, 2, 2, 3, 3, 4] )
        >>> True

        Scheduling(7).feasible( [6, 5, 1],  [0, 1, 2, 2, 3, 3, 4] )
        >>> False

        """
        self.tmp = profit_jobs
        self.is_feasible = True

        i = 0
        j = 1
        k = 0

        while i < len(self.tmp):
            while j < len(self.tmp):
                self.index1 = self.jobs.index(self.tmp[i])
                self.index2 = self.jobs.index(self.tmp[j])
                j += 1
                if deadline[self.index1] > deadline[self.index2]:
                    (self.tmp[i], self.tmp[j]) = (
                        self.tmp[j],
                        self.tmp[i],
                    )
            i += 1

        while k < len(self.tmp):
            self.job = self.tmp[k]
            self.jobindex = self.jobs.index(self.job)
            self.dlineval = deadline[self.jobindex]
            self.ftest = k + 1
            k += 1
            if self.dlineval < self.ftest:
                self.is_feasible = False
                break
        return self.is_feasible


def main():
    job = namedtuple("job", "job_id deadline profit")
    jobs = [
        job(0, 0, 0),
        job(1, 2, 46),
        job(2, 4, 52),
        job(3, 3, 30),
        job(4, 3, 36),
        job(5, 2, 56),
        job(6, 1, 40),
    ]
    # midresult stores jobs in sorting order of deadline
    midresult = []
    for i in range(len(jobs)):
        current_job = []
        current_job.extend((jobs[i].deadline, jobs[i].profit, jobs[i].job_id))
        midresult.append(current_job)
    midresult.sort(key=lambda k: (k[0], -k[1]))
    (deadline, profit, jobs) = map(list, zip(*midresult))

    scheduling_jobs = Scheduling(jobs)
    scheduled_jobs = scheduling_jobs.schedule(len(jobs), deadline)
    print(f"\n Jobs {scheduled_jobs}")

    finalprofit = []
    finaldl = []

    for i, item in enumerate(scheduled_jobs):
        jobsindex = jobs.index(item)
        finalprofit.append(profit[jobsindex])
        finaldl.append(deadline[jobsindex])

    print(f"\n Profit {finalprofit}")
    print(f"\n Deadline {finaldl}")


if __name__ == "__main__":
    main()
```

**Comment by ichard26** [comment]
Hi, this is **not** a bug within Black, this is actually a bug of the Black Playground (i.e. https://black.now.sh) caused by https://github.com/jpadilla/black-playground/commit/7a81bc1f9c5dff342e01ec52734a1c493f7a004e where it wasn't handling `black.NothingChanged` exceptions properly. When the source code and the formatted code is the same, the server processing your requests thought Black couldn't format your code and then tried to return the error message. There is no output since `black.NothingChanged` is raised without an error message argument. 

I've already worked on a fix (https://github.com/jpadilla/black-playground/pull/35) so now we play the waiting game :) Thanks for reporting anyway, if this was a bug **within** Black that wasn't caught by the automated testing, we would have had a really bad situation on our hands :-).

**Comment by ichard26** [comment]
I have no idea whether I messed up something or something went wrong with GitHub's autoclosing upon merge feature but this is now resolved. The example provided above no longer causes the Black Playground to return nothing.

**Comment by Mohitkumar6122** [comment]
> Hi, this is **not** a bug within Black, this is actually a bug of the Black Playground (i.e. https://black.now.sh) caused by [jpadilla/black-playground@7a81bc1](https://github.com/jpadilla/black-playground/commit/7a81bc1f9c5dff342e01ec52734a1c493f7a004e) where it wasn't handling `black.NothingChanged` exceptions properly. When the source code and the formatted code is the same, the server processing your requests thought Black couldn't format your code so it tried to return the error message. There is no output since `black.NothingChanged` is raised without an error message argument.
> 
> I've already worked on a fix ([jpadilla/black-playground#35](https://github.com/jpadilla/black-playground/pull/35)) so now we play the waiting game :) Thanks for reporting anyway, if this was a bug **within** Black that wasn't caught by the automated testing, we would have had a really bad situation on our hands :-).

thanks now i get it ...
